src/main/java/com/coveros/selenified/element/check/azzert/AssertContains.java

```diff
@@ -0,0 +1,245 @@
+    // assessing functionality
+    // ///////////////////////////////////////
+
+    /**
+     * Asserts that the element's class contains the provided expected class.
+     * If the element isn't present, this will constitute a failure, same as a
+     * mismatch. This information will be logged and recorded, with a screenshot
+     * for traceability and added debugging support.
+     *
+     * @param expectedClass - the expected class value
+     */
+    public void clazz(String expectedClass) {
+        String clazz = checkClazz(expectedClass, 0, 0);
+        assertNotNull(NO_ELEMENT_FOUND, clazz);
+        assertTrue("Class Mismatch: class of '" + clazz + DOES_NOT_CONTAIN + expectedClass + "'", clazz.contains(expectedClass));
+    }
+
+    /**
+     * Asserts that the element contains the provided expected attribute. If
+     * the element isn't present, this will constitute a failure, same as a
+     * mismatch. This information will be logged and recorded, with a screenshot
+     * for traceability and added debugging support.
+     *
+     * @param expectedAttribute - the attribute to check for
+     */
+    public void attribute(String expectedAttribute) {
+        String[] attributes = checkAttribute(expectedAttribute, 0, 0);
+        assertNotNull(NO_ELEMENT_FOUND, attributes);
+        assertTrue("Attribute not found: element attributes of '" + String.join(",", attributes) +
+                DOES_NOT_CONTAIN + expectedAttribute + "'", Arrays.asList(attributes).contains(expectedAttribute));
+    }
+
+    /**
+     * Asserts that the element's text contains the provided expected text. If
+     * the element isn't present, this will constitute a failure, same as a
+     * mismatch. This information will be logged and recorded, with a screenshot
+     * for traceability and added debugging support.
+     *
+     * @param expectedText the expected value of the element
+     */
+    public void text(String expectedText) {
+        String text = checkText(expectedText, 0, 0);
+        assertNotNull(NO_ELEMENT_FOUND, text);
+        assertTrue("Text not found: element text of '" + text + DOES_NOT_CONTAIN + expectedText + "'", text.contains(expectedText));
+    }
+
+    /**
+     * Asserts that the element's value contains the provided expected value.
+     * If the element isn't present or an input, this will constitute a failure,
+     * same as a mismatch. This information will be logged and recorded, with a
+     * screenshot for traceability and added debugging support.
+     *
+     * @param expectedValue the expected value of the element
+     */
+    public void value(String expectedValue) {
+        String value = checkValue(expectedValue, 0, 0);
+        if (value == null) {
+            String reason = NO_ELEMENT_FOUND;
+            if (getElement().is().present()) {
+                reason = "Element not input";
+            }
+            fail(reason);
+        }
+        assertNotNull(value);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "value" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS2259)